graphistry/arrow_uploader.py

```diff
@@ -380,7 +380,17 @@ def verify(self, token=None) -> bool:
             json={'token': token})
         return out.status_code == requests.codes.ok
 
-    def create_dataset(self, json):  # noqa: F811
+    def validate_encoding(self, json):
+        edge_attribute = json['edge_encodings']['complex']['default']['edgeColorEncoding']['attribute']
+        edge_attributes = self.edges.column_names
+        if not edge_attribute in edge_attributes:
+            err = f'Invalid edge encoding: attribute {edge_attribute} does not exists in {str(edge_attributes)}'
+            logger.error(err)
+            raise Exception(err)
+
+    def create_dataset(self, json, validate: bool = True):  # noqa: F811
+        if validate:
+            self.validate_encoding(json)
```

Comment:
I'd think:
-  we move the ~entire FEP validator in here (and keep there too for now)
-  add a pass for the new checks introduced here that check against the node/edge tables

Reply:
AFAICT we can move all of them, making this PR about:

- `g.validate()`: Entrypoint that, for now, validate metadata -- bindings, simple encodings, complex encodings, settings, ... 
  - And motivating all this, when `._nodes` / `._edges` is available, against that schema. This should be skippable somehow, so that FEP can switch to using PyGraphistry, and run validate even tho the tables aren't available.
- `g.plot(..., validate=True)`: Running that (default-on) for api=3 uploads

Let's defer dataset validations for another PR, e.g., null ID value checks, as that's a big item to itself

Reply:
Awesome, that makes sense.  I'll move the logic from `FEP/validate_encodings.py`; I'll create another file (`validate_encodings.py`) just to keep clean the `arrow_uploader.py` module.

Reply:
Done!